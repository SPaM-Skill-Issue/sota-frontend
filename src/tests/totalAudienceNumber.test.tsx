import { render, screen, waitFor } from "@testing-library/react";
import TotalAudienceNumber from "../components/totalAudienceNumber";
import "@testing-library/jest-dom";
import { MemoryRouter } from "react-router-dom";
import {
    mockAudienceData,
    totalNumberOfAudient,
} from "./mocks/mockAudienceData";

describe('TotalAudienceNumber Component Tests', () => {
    it ("should render TotalAudienceNumber component correctly", () => {
        const wrapper = render(<TotalAudienceNumber fetch_data={[]} />);
        expect(wrapper).toMatchSnapshot();
    });

    it ("should displays loading state initially", async () => {
        render(<TotalAudienceNumber fetch_data={mockAudienceData} />);
        await waitFor(() => {
            const element = screen.getByText(/Total Number of Audience/i);
            expect(element).toBeInTheDocument();
        });
    });

    it ("should display total audience number correctly", async () => {
        render(
            <MemoryRouter>
                <TotalAudienceNumber fetch_data={mockAudienceData} />
            </MemoryRouter>
        );
        await waitFor(() => {
            const totalSpan = screen.getByTestId("total");
            expect(totalSpan).toHaveTextContent(String(totalNumberOfAudient["All"]));
            expect(totalSpan).toBeInTheDocument();
        });
    });

    it ("should display total male audience number correctly", async () => {
        render(
            <MemoryRouter>
                <TotalAudienceNumber fetch_data={mockAudienceData} />
            </MemoryRouter>
        );
        await waitFor(() => {
            const totalSpan = screen.getByTestId("total-m");
            expect(totalSpan).toHaveTextContent(String(totalNumberOfAudient["M"]));
            expect(totalSpan).toBeInTheDocument();
        });
    });

    it ("should display total female audience number correctly", async () => {
        render(
            <MemoryRouter>
                <TotalAudienceNumber fetch_data={mockAudienceData} />
            </MemoryRouter>
        );
        await waitFor(() => {
            const totalSpan = screen.getByTestId("total-f");
            expect(totalSpan).toHaveTextContent(String(totalNumberOfAudient["F"]));
            expect(totalSpan).toBeInTheDocument();
        });
    });

    it ("should display total not identify gender audience number correctly", async () => {
        render(
            <MemoryRouter>
                <TotalAudienceNumber fetch_data={mockAudienceData} />
            </MemoryRouter>
        );
        await waitFor(() => {
            const totalSpan = screen.getByTestId("total-n");
            expect(totalSpan).toHaveTextContent(String(totalNumberOfAudient["N"]));
            expect(totalSpan).toBeInTheDocument();
        });
    });
});