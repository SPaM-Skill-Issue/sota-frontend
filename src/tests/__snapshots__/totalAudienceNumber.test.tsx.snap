// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalAudienceNumber Component Tests should render TotalAudienceNumber component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=" bg-belft-blue rounded-2xl"
      >
        <div
          class=" p-5"
        >
          <span
            class=" font-primary text-2xl text-white"
          >
            Total Number of Audient
          </span>
          <div
            class="flex items-center justify-center w-full h-[20vh]"
          >
            <span
              class="font-primary text-hunyadi-yellow text-3xl"
            >
              No DATA
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=" bg-belft-blue rounded-2xl"
    >
      <div
        class=" p-5"
      >
        <span
          class=" font-primary text-2xl text-white"
        >
          Total Number of Audient
        </span>
        <div
          class="flex items-center justify-center w-full h-[20vh]"
        >
          <span
            class="font-primary text-hunyadi-yellow text-3xl"
          >
            No DATA
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
